### ISSUE TITLE
Enhancement: pre-cache picture thumbnails 

### ISSUE THREAD

**Issue by Gatak** [nextcloud/android#4496]
Hi. 

I would like the option that NC android client can pre cache the thumbnails for certain folders. This would make the user experience much better. In my work flow I import many photos from my DSLR camera on the desktop, but I would like to view and share them on the phone. Even if I to occ preview:generate-all, the download speed of thumbnails is too slow to do on-demand only.

**Comment by tobiasKaminsky** [comment]
I had something similar in mind, but having an option for that is too techy, in my opinion.
Better would be something like:
- have a background job that runs every 15 min when on wifi (to not consume mobile data)
- get a list of changed images/previewable files 
- check for outdated thumbnails
- download them

This would then also require that cache might increase beyond 200mb (which is currently hard limited, but I guess in modern devices we can use a better max value, eg. based on free disk space, etc.)

**Comment by tobiasKaminsky** [comment]
Same would also be true for entire file structure, when assuming that ~80% of files are changed very very rarely on private instances.

**Comment by Gatak** [comment]
Yes. Of course. But syncing all thumbs by default might take to much storage space on the phone if the user has many files. But having this as an option would be very valuable.

**Comment by AndyScherzinger** [comment]
If we simply pre-fecth based on the loader logic of the photos-view we could do such a thing which basically translates to loading the latest n image thumbnails.

**Comment by Gatak** [comment]
Not sure I follow. "latest n image thumbnails". Do you mean to schedule downloading small sets odd thumbs in the background?

**Comment by Whisprin** [comment]
Found this old issue that seems related: https://github.com/nextcloud/android/issues/177

I think photos and videos are one of the main use cases for people on their phones. Maybe it makes sense to give a bit more control to the user about keeping smaller versions(/thumbnails) of their media on the phone as suggested here:
https://github.com/nextcloud/android/issues/1490

The logic for the use cases may be very similar.

**Comment by acloudadmin** [comment]
Hi. Great app -- very useful.

Like to add my support for improving the thumbnail cache. I have 15k photos and lots and lots of room on my phone - more than enough room to cache thumbnails for all. This would make the app SOOO much more useful to me.

If I had the option, I 'd love to be able to set my own cache-size limit, and perhaps even prot CT the cache from being cleaned out by any routine android cache-cleaning protocols (of such things exist).

Thanks again for all your work on this.

**Comment by tobiasKaminsky** [comment]
Regarding cache size: #596
I know it is outdated, but maybe we can research if an unlimited cache size is possible.
(Android allows to clean cache, if device runs out of space, so we need not to limit it arbitrary)

**Comment by axheli** [comment]
@tobiasKaminsky any progress on this ?

**Comment by FlashyDuck** [comment]
This is really a feature that kills the experience. I've transferred a few users from icloud to my nextcloud server and this is the biggest complaint that I got. Although I have Previewgenerator installed and all, but still; the app only loads/downloads thumbnails that the user is viewing. Large folders to scroll through (1000+ pictures) is a real pain.

**Comment by shaun-ba** [comment]
I agree this is just stupid, give the user control of this...

I cannot even view a pretty small folder of photos, the client says 78mb used and has a red bar at the bottom. Naturally I went to settings to increase the cache and was shocked to find that isn't an option.
